resources/lib/api.py

```diff
@@ -195,6 +212,23 @@ def create_session(self, refresh=False) -> None:
         )
         account_data.update(r)
 
+        # @todo: this will call the profiles frequently, we should cache that
+        r = self.make_request(
+            method="GET",
+            url=self.PROFILES_LIST_ENDPOINT,
+        )
+
+        if action == "refresh_profile":
+            account_data.update(
+                next(profile for profile in r.get("profiles") if profile["profile_id"] == account_data["profile_id"]))
+        else:
```

Comment:
I'm a bit unsure what this part is supposed to do. Do we need it? Because if not, we can save ourselves the API call in 216.
As far as I can tell, it loads the account data from cache (which is already done in 103, `profile_id` can then be found in `self.profile_id`), then it checks whether it has that id, then compares that against the profiles from crunchyroll and if so, update the account data with the crunchyroll data - but we should already have that, since it was loaded from cache?
What am I'm missing? :)

Reply:
Hm I think I get it. But it remains suboptimal, since we do a lot of unnecessary requests to the multiprofile endpoint. I'll refactor that, but still thinking about the best way to solve that.